### ISSUE TITLE
W3C Schema parser broken for most facets and constraints

### ISSUE THREAD

**Issue by GoogleCodeExporter** [sshyran/genxdm#63]
```
As a part of removing the <A> parameter from schema model, the Schema Parser's 
use of AtomBridge was removed.  The schema parser implementation used the atom 
bridge to validate the values of various constraints.

First: it is questionable whether this is the right approach.  A supplied atom 
bridge may not behave correctly.  Should the behavior of the schema parser be 
that variable and caller-defined?  I doubt that that was the intent.

Second: it can be fixed by creating a canonical atom bridge, used only by the 
schema parser.  This may be the least extensive change to make.

Third: it can be fixed by implementing the validation functionality of atom 
bridge types without having an atom bridge.

Fourth: the biggest problem is schema constraint checker, which is a procedural 
library for validating schema.  As designs go, procedural code in java is 
perhaps my least favorite.

This issue is marked blocked by 13.  It only exists if the solution to 13 is 
accepted, and merged into trunk.  At that point, it becomes rather high 
priority.
```

Original issue reported on code.google.com by `aale...@gmail.com` on 6 Jun 2011 at 3:28
- Blocking: #64
- Blocked on: #13

**Comment by GoogleCodeExporter** [comment]
Original comment by `aale...@gmail.com` on 7 Jun 2011 at 12:30

**Comment by GoogleCodeExporter** [comment]
```
Okay, in progress.  Created a 'canonical' atom bridge for use by the schema 
parser.  completed rework of XMLSchemaParser.  Still have to do the converter 
and the constraint checker.  The canonical bridge is mostly placeholder at the 
moment; we'll get it in proper shape shortly.
```

Original comment by `aale...@gmail.com` on 7 Jun 2011 at 4:43
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
Several methods in CanonicalAtomBridge remain, together with restoring the 
functionality of the SchemaConstraintChecker (though if I can de-library the 
damned thing I will).
```

Original comment by `aale...@gmail.com` on 7 Jun 2011 at 9:23

**Comment by GoogleCodeExporter** [comment]
```
This should be fixed as of r236, but I can't prove it, 'cause the tests aren't 
written (though there's an infrastructure in place).  The technique for unit 
testing: instantiate a parser, hand it an input stream for a known schema, then 
examine the iterables in the returned component bag to verify that we see what 
we expect, and that there's nothing there we don't expect.  Further tests would 
include more complex schemas (with includes and imports and customized 
resolvers and catalogs and things) and failure modes.
```

Original comment by `aale...@gmail.com` on 9 Jun 2011 at 7:21

**Comment by GoogleCodeExporter** [comment]
```
fixed a while ago, confirmed-ish by the schema tests that were committed 
earlier.
```

Original comment by `aale...@gmail.com` on 29 Aug 2011 at 6:53
- Changed state: **Fixed**
